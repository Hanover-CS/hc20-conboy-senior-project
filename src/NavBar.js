/* Author: Patrick Conboy
 * Date Created: 12/22/2019
 * Description: This file creates the static navigation bar that
 * sits at the top of the AniTracker website.
 */

import React, { Component } from 'react'
import { Route, NavLink, Switch, BrowserRouter as Router } from 'react-router-dom'

import './css/NavBar.css'
import SearchPage from './SearchPage'
import Home from './Home'
import AnimeList from './AnimeList'

class NavBar extends Component {
  render() {
    return (
      <Router>
        <div className='NavBar'>

          <ul id='navLinks'>
            <li>
              <NavLink to='/'>Home</NavLink>
            </li>
            <li>
              <NavLink to='/search-page'>Search Page</NavLink>
            </li>
            <li>
              <NavLink to='/anime-list'>Anime List</NavLink>
            </li>
          </ul>

          <Switch>
            <Route exact path="/" component={Home}/>
            <Route path="/search-page" component={SearchPage}/>
            <Route path="/anime-list" component={AnimeList}/>
          </Switch>

        </div >
      </Router>
    )
  }
}

export default NavBar;