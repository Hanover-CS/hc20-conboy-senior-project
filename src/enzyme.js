/* Author: Patrick Conboy
 * Date Created: 1/1/2019
 * Description: This file initializes Enzyme for testing purposes.
 */

import Enzyme, { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
export { shallow, mount, render };
export default Enzyme;