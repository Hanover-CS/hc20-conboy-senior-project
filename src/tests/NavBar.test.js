import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from '../enzyme'

import NavBar from '../NavBar'

describe('NavBar Tests', () => {
  test('nav bar renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<NavBar />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('nav bar renders correctly', () => {
    const wrapper = shallow(<NavBar />)

    expect(wrapper.find('#NavBar')).toBeDefined()
    expect(wrapper.find('#navLinks')).toBeDefined()
  })

  test('snapshot of nav bar hasn\'t changed', () => {
    const tree = renderer.create(<NavBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})