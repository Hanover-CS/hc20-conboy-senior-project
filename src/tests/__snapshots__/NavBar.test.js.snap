// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Tests snapshot of nav bar hasn't changed 1`] = `
<div
  className="NavBar"
>
  <ul
    id="navLinks"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/search-page"
        onClick={[Function]}
      >
        Search Page
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/anime-list"
        onClick={[Function]}
      >
        Anime List
      </a>
    </li>
  </ul>
  <div
    className="HomePage"
  >
    <h3>
      Welcome to AniTracker!
    </h3>
  </div>
</div>
`;
